draft-ietf-tls-esni.md

```diff
@@ -699,24 +699,30 @@ MAY offer to resume sessions established without ECH.
 ## Client-Facing Server
 
 Upon receiving an "encrypted_client_hello" extension, the client-facing server
-MUST check that it is able to negotiate TLS 1.3 or greater. If not, it MUST
-abort the connection with a "handshake_failure" alert.
+first checks that it is able to negotiate TLS 1.3 or higher. If not, then it
+MUST ignore the extension and proceed with the handshake as usual. Next, the
+server checks that the ClientHello contains a "server_name" extension and that
+the indicated name is the public name of at least one of its ECH configurations.
+If not, then the server MUST ignore the "encrypted_client_hello" extension and
+proceed as usual. (Presumably the extension was generated as specified in
+{{grease-extensions}} and not intended as a real ECH offer.) Otherwise, the
```

Comment:
That's not true. We may also just have a key mismatch, which was the original reason all this stuff was added. For instance, maybe the client picked up a very stale key from the DNS, or this server didn't get a new key config despite it being published in the DNS already.